### PR TITLE
Auto generate ecalls FFI rust code using bindgen in enigma-core app

### PR DESCRIPTION
**Pull request by elichai**
This auto generates the FFI bindings in the app's `build.rs` file.

### CODE REVIEW

File: enigma-core/app/Cargo.toml [hunk 1]
```diff
@@ -19,7 +19,8 @@ failure = "0.1.3"
 rustc-hex = "1.0.0" # 2.0.1?
 dirs = "1.0.4"
 # TODO: Add compression as a feature and use it in `set_compression_type()`
-rocksdb = { version = "0.12.2", default-features = false } # TODO: Update the rocksdb lib https://github.com/rust-rocksdb/rust-rocksdb/issues/304
+#rocksdb = { version = "0.12.2", default-features = false }
+rocksdb = { git = "https://github.com/rust-rocksdb/rust-rocksdb.git", rev = "067c72a", default-features = false } # replace when there's a new version.
```
Comment (AvishaiW) [inline enigma-core/app/Cargo.toml:23]: :)

File: enigma-core/app/build.rs [hunk 2]
```diff
@@ -19,4 +25,33 @@ fn main() {
             println!("cargo:rustc-link-lib=dylib=sgx_uae_service");
         }
     }
+
+    let edl = format!("{}/edl", rust_sgx_sdk);
+
+    let bindings = builder()
```
Comment (AvishaiW) [inline enigma-core/app/build.rs:31]: it would be helpful if you can add documentation to this.

**Review by moriaab — APPROVED** [review]
Great addition to the core code!

File: enigma-core/app/build.rs (hunk 2 above)
Comment (elichai) [reply enigma-core/app/build.rs:31]: You're right. this will require more docs, I'll do that after I merge `bindgen` all over if I have the time.

**Pull request merged**